internal/plugins/golang/api.go

```diff
@@ -52,5 +63,21 @@ func (p *createAPIPlugin) Run() error {
 
 // SDK plugin-specific scaffolds.
 func (p *createAPIPlugin) run() error {
+	// Write CR paths to the samples' kustomization file. This file has a
+	// "do not modify" comment so it is safe to overwrite.
+	samplesKustomization := sampleKustomizationFragment
+	for _, gvk := range p.config.Resources {
+		samplesKustomization += fmt.Sprintf("- %s\n", makeCRFile(gvk))
+	}
+	kpath := filepath.Join("config", "samples")
+	if err := kustomize.Write(kpath, samplesKustomization); err != nil {
+		return err
+	}
```

Comment:
Should we really re-write the kb plugin? Would not be better we add a new our/plugin that will be executed after just to scaffold its requirement? something as ` (p *createBundlePlugin)`. If we follow up this approach can we not end up in a situation where we will change the kb plugins and then, both projects are no longer compatible?

Reply:
>add a new our/plugin that will be executed after just to scaffold its requirement

If I understand what you're asking, we are effectively doing that with:

```go
import (
	...
	"sigs.k8s.io/kubebuilder/pkg/plugin"
)

type createAPIPlugin struct {
	plugin.CreateAPI
	...
}

func (p *createAPIPlugin) Run() error {
	if err := p.CreateAPI.Run(); err != nil {
		return err
	}
	...
	return p.run()
}
```

This emulates phase 2 plugin behavior.

Reply:
My idea here was that we would have a new plugin such as  `_ sdkplugin.BundlePluginGetter       = Plugin{}` and not overwrite the init one.  And then, the Bundle plugin when/if called, would update the Makefile scaffolds and etc which means that we could indeed just to do the customization if a command such as  `operator-sdk bundle create` be executed for example. 

**However, I am not sure if it will be easy to be done now .. so it is fine. Out of scope here.**  👍